test/extensions/tracers/xray/fuzz_test.cc

```diff
@@ -0,0 +1,22 @@
+#include "extensions/tracers/xray/util.h"
+
+#include "test/fuzz/fuzz_runner.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Tracers {
+namespace XRay {
+
+DEFINE_FUZZER(const uint8_t* buf, size_t len) {
+  absl::string_view sv(reinterpret_cast<const char*>(buf), len);
+  wildcardMatch("*", sv);
```

Comment:
Can you split the incoming buffer so that you pivot on the first few bytes of randomness as an index, allowing some of the buffer to be the wildcard pattern and the rest the lookup term? That would simplify this down to a single `wildcardMatch`.